### PR TITLE
Added provision for automate ping test under performanceTest in

### PR DESCRIPTION
**Pull request by bhavikkothari-eaton**
Added provision to automate ping test under performanceTest for non-docker mode

### CODE REVIEW

**Comment by rajatkumar-eaton** [comment]
@bhavikkothari-eaton  Can you please list down the performance test cases(from EdgeX- TAF) we are covering against this PR, just want to see if something already there in our pytest repo ?

File: prerequisite_changes.py [hunk 1]
```diff
@@ -228,7 +232,53 @@ def main():
         replace("TAF/testScenarios/functionalTest/V2-API/core-command/device/GET-Positive.robot",
                 "== 6",
                 "== ${length}")
+    if service == "4_ping_response_time" or service=="performance-metrics-collection":
```
Comment (rajatkumar-eaton) [inline prerequisite_changes.py:235]: Is this the name of ping service in TAF? @bhavikkothari-eaton

File: prerequisite_changes.py [hunk 2]
```diff
@@ -228,7 +232,53 @@ def main():
...
+        # Added dependent library in test robot file
+        add_data("TAF/testScenarios/performanceTest/performance-metrics-collection/4_ping_response_time/ping_response_time.robot", 
+                 "Resource        TAF/testCaseModules/keywords/common/commonKeywords.robot\n",
+                 "Variables         TAF/config/performance-metrics/configuration.py\n")
+        # Remove service ping as service is not available in swagger but was available in docker container
```
Comment (rajatkumar-eaton) [inline prerequisite_changes.py:240]: ping api is already available in our edgex also for all the services, we have that API exposed in Swagger page.

Comment (rajatkumar-eaton) [reply prerequisite_changes.py:240]: I want to discuss offline also.

Comment (rajatkumar-eaton) [reply prerequisite_changes.py:240]: We are already doing ping test for all the services in our smoke test file.

https://github.com/etn-ccis/edge-linux-test-edgex-pytest/blob/e2c90729943b6b57cd4298927e85156fc3cea413/px_red/tests/edgex/test_edgex_smoke.py#L114

`    @pytest.mark.parametrize("api_gateway_service", test_data["api_gateway_services"])
    def test_ping_edgex_services(self, api_gateway_service):
        self.console.edgex_client.check_services(service_list)`

File: prerequisite_changes.py (hunk 1 above)
Comment (bhavikkothari-eaton) [reply prerequisite_changes.py:235]: It is the **ping_response_time.robot** test which measure the ping response time of ping API for each edgex service.
service list 
1. core-data
2. core-metadata
3. core-command
4. support-scheduler
5. support-notification
6. sys--mgmt-agent
7. device-rest
8. device-rest
9. app-service
10. device-virtual

**Comment by bhavikkothari-eaton** [comment]
List of tests for performance test in Edgex-TAF

1) retrieve_systeminfo_info
-This test fetch system related information like CPU frequency, system memory 
2) retrieve_footprint
-There is only one testcase under this robot file and it fetching docker container imageby service name.
3) service_startup_time
-There are 2 testcase under this robot file
   1>Get service startup time with creating containers-This will start 
      dicker and check service up time
   2>Get service startup time without creating containers-This will 
      directly check service up time in docker containers.
4) resource_usage_with_autoevent
  -This  test is checkig threshold value of CPU and memory for each 
    container image.
5) ping_response_time
   -This test is pinging each service available in docker container
6) event_exported_time
  -This test checking threshold time
7) summary_reports
  - This test case creating reports of above tests with creating 
     containers and without creating containers

**Review by rajatkumar-eaton — APPROVED** [review]

**Pull request merged**